clang/include/clang/Basic/TokenKinds.h

```diff
@@ -95,10 +95,7 @@ inline bool isStringLiteral(TokenKind K) {
 /// Return true if this is a "literal" kind, like a numeric
 /// constant, string, etc.
 inline bool isLiteral(TokenKind K) {
-  return K == tok::numeric_constant || K == tok::char_constant ||
-         K == tok::wide_char_constant || K == tok::utf8_char_constant ||
-         K == tok::utf16_char_constant || K == tok::utf32_char_constant ||
-         isStringLiteral(K) || K == tok::header_name || K == tok::binary_data;
+  return K >= tok::numeric_constant && K <= tok::utf32_string_literal;
```

Comment:
Can we add a comment in `TokenKinds.def`  (before numeric_constant) such that we guarantee someone does not change the order or add an assert that tests for equality?
I don't think we currently rely on, or guarantee an order here @AaronBallman